tuify/examples/main_interactive.rs

```diff
@@ -15,223 +15,329 @@
-        }
-        None => println!("User did not select anything"),
-    }
-    call_if_true!(TRACE, {
-        log_debug(format!("user_input: {:?}", user_input).to_string());
-    });
-}
+ use std::io::Result;
+
+ use r3bl_ansi_color::{AnsiStyledText, Color as RColor, Style as RStyle};
+ use r3bl_rs_utils_core::*;
+ use r3bl_tuify::*;
+
+ fn print_header(msg: &str) {
+     AnsiStyledText {
+         text: msg,
+         style: &[
+             RStyle::Bold,
+             RStyle::Italic,
+             RStyle::Underline,
+             RStyle::Foreground(RColor::Rgb(236, 230, 230)),
+             RStyle::Background(RColor::Rgb(10, 109, 33)),
+         ],
+     }
+     .println();
+ }
+
+ fn main() -> Result<()> {
+     throws!({
+         call_if_true!(TRACE, {
+             try_to_set_log_level(log::LevelFilter::Trace).ok();
+             log_debug("Start logging...".to_string());
+             log_debug(format!("og_size: {:?}", get_size()?).to_string());
+         });
+
+         // Get display size.
+         let max_width_col_count: usize =
+             get_size().map(|it| it.col_count).unwrap_or(ch!(80)).into();
+         let max_height_row_count: usize = 5;
+
+         let style = StyleSheet::default();
+
+         let up_down_colored = AnsiStyledText {
+             text: &format!("{}", "`Up` or `Down` arrow",),
+             style: &[RStyle::Foreground(RColor::Rgb(0, 255, 0))],
+         };
+
+         let space_colored = AnsiStyledText {
+             text: &format!("{}", "`Space`",),
+             style: &[RStyle::Foreground(RColor::Rgb(255, 216, 9))],
+         };
+
+         let esc_colored = AnsiStyledText {
+             text: &format!("{}", "`Esc`",),
+             style: &[RStyle::Foreground(RColor::Rgb(255, 132, 18))],
+         };
+
+         let return_colored = AnsiStyledText {
+             text: &format!("{}", "`Return`",),
+             style: &[RStyle::Foreground(RColor::Rgb(234, 0, 196))],
+         };
+
+         let multi_select_instructions = AnsiStyledText {
+             text: &format!(
+                 "{}{up_down_colored}{}{space_colored}{}{esc_colored}{}{return_colored}{}",
+                 "",
+                 " --> To move up or down\n",
+                 " --> To Select or deselect branches\n",
+                 " --> To exit\n",
+                 " --> To confirm selection"
+             ),
+             style: &[RStyle::Foreground(RColor::Rgb(195, 0, 94))],
+         };
+
+         // Add tuify to select which example to run.
+         let maybe_user_input = select_from_list(
+             "Select which example to run".to_string(),
+             [
+                 "Multiple select, single item",
+                 "Multiple select, 13 items, viewport height = 5",
+                 "Multiple select, 2 items, viewport height = 5",
+                 "Single select, 13 items, viewport height = 5",
+                 "Single select, 2 items, viewport height = 5",
```

Comment:
Since these string literals are used below, maybe put each one in a string constant.

Something like:
`const MS_SI:&str = "Multiple select, single item;`

eg: https://github.com/r3bl-org/r3bl-open-core/blob/main/ansi_color/src/ansi_escape_codes.rs#L52

Reply:
Great idea, thanks, done!